spec/features/settings/dex_connector_ldap_feature_spec.rb

```diff
@@ -130,6 +130,75 @@
     end
   end
 
+  # rubocop:disable RSpec/ExpectInHook
+  describe "#new_save_disabled_by_change" do
+    before do
+      visit new_settings_dex_connector_ldap_path
+      expect(page.evaluate_script("$('#ldap_conn_save').prop('disabled')")).to eq(true)
+
+      # Test that we can enable the button
+      page.execute_script("$('#ldap_conn_save').removeProp('disabled')")
+      expect(page.evaluate_script("$('#ldap_conn_save').prop('disabled')")).to eq(false)
+    end
+
+    it "allows change of start_tls" do
+      # Is false by default - change to true
+      # Following line should work but it does not so using JS
+      # choose id: "dex_connector_ldap_start_tls_true"
+      page.execute_script("$('#dex_connector_ldap_start_tls_true').prop('checked', true)")
+      page.execute_script("$('#dex_connector_ldap_start_tls_true').trigger('change')")
+    end
+
+    it "allows change of bind_anon" do
+      # Is false by default - change to true
+      # Following line should work but it does not so using JS
+      # choose id: "dex_connector_ldap_bind_anon_true"
+      page.execute_script("$('#dex_connector_ldap_bind_anon_true').prop('checked', true)")
+      page.execute_script("$('#dex_connector_ldap_bind_anon_true').trigger('change')")
+    end
+
+    it "allows change of host" do
+      fill_in "Host", with: "AAA"
+      page.execute_script("$('#dex_connector_ldap_host').trigger('change')")
+    end
+
+    it "allows change of port" do
+      fill_in "Port", with: "AAA"
+      page.execute_script("$('#dex_connector_ldap_port').trigger('change')")
+    end
+
+    it "allows change certificate" do
+      attach_file "Certificate", admin_cert_file.path
+      page.execute_script("$('#dex_connector_ldap_certificate').trigger('change')")
+    end
+
+    it "allows change of bind dn" do
+      fill_in id: "dex_connector_ldap_bind_dn", with: "AAA"
+      page.execute_script("$('#dex_connector_ldap_bind_dn').trigger('change')")
+    end
+
+    it "allows change of password" do
+      page.execute_script("$('#dex_connector_ldap_bind_pw[required=\"required\"]').val('AAA')")
```

Comment:
You can use the `fill_in` matcher here using `dex_connector_ldap_bind_pw` as the locator instead of `execute_script`. 

Even having duplicated elements with the same id, `fill_in` will try to find the visible elements and will ignore the duplicated one. This will also benefit the normal engine behavior when filling the input text to avoid unexpected results.

Reply:
Since fill_in does not properly trigger onchange events, which are needed for this test, the exact same code you want to remove using jquery to grab the element still has to exist on the next line. Making a change like this as a result is rather pointless. I changed it and have tested it and will push the update, but I don't think this is a good use of my time, anyone else's time, nor of the test servers time retesting this for an inconsequential change.

"Avoid unexpected results"? What are you talking about? We intend to stop using Ruby and Rails and Capybara in the future entirely so that is an irrelevant optimization.